lib/srv/db/common/auth.go

```diff
@@ -381,15 +381,18 @@ func (a *dbAuth) getTLSConfigVerifyFull(ctx context.Context, sessionCtx *Session
 
 // getTLSConfigInsecure generates tls.Config when TLS mode is equal to 'insecure'.
 // Generated configuration will accept any certificate provided by database.
-func getTLSConfigInsecure() *tls.Config {
-	tlsConfig := &tls.Config{
-		RootCAs: x509.NewCertPool(),
+func (a *dbAuth) getTLSConfigInsecure(ctx context.Context, sessionCtx *Session) (*tls.Config, error) {
+	tlsConfig, err := a.getTLSConfigVerifyFull(ctx, sessionCtx)
```

Comment:
Do we really need to build a full TLS verification config first if we're going to disable verification anyway? Wouldn't just adding "appendClientCert()" work?

Reply:
We probably could, but then we have different behaviour for cloud dbs. Currently we don't include the client certificate in other modes: https://github.com/gravitational/teleport/blob/bfaf3f8228cdfa88febbd0fba094488cbda601f1/lib/srv/db/common/auth.go#L372-L374

Reply:
Actually, you're right, but it seems like that below we're adding client cert in any case? Even when connecting to a cloud database with "insecure" flag set?